fix(ci): build core exports before real Scenario E2E API harnesses

## Problem

`Scenario E2E` on current `develop@0fa327137428247add4e2f98d98e78bb3f97e50d` starts real local API harnesses without first building the workspace packages they import.

Two independent jobs fail at startup:

- `Zero-Key unit + UI coverage` -> `Real local chat pipeline (no model key, no llama)`
- `Zero-Key accounts UI e2e (real API + pool + disk)` -> accounts API server

Both resolve source that imports built `@elizaos/core` subpaths, but the setup step only builds private packages. Linux then fails with either:

```text
Cannot find module '@elizaos/core/client-public'
```

or:

```text
Cannot find .../plugins/plugin-scheduling/node_modules/@elizaos/core/dist/edge/index.edge.js
```

Exact run: https://github.com/elizaOS/eliza/actions/runs/31927388847

## Acceptance criteria

- Build the narrow required workspace graph before both real harnesses.
- Keep the deterministic zero-key contract; do not add provider secrets or replace the system under test with mocks.
- Reproduce both failing commands locally from a clean/build-equivalent state.
- Add or update a workflow contract test so the build ordering cannot regress.
- Run focused tests and `bun run verify`; confirm the exact-head hosted Scenario E2E jobs pass.

--- a/.github/workflows/scenario-pr.yml
+++ b/.github/workflows/scenario-pr.yml
@@ -85,6 +85,9 @@ jobs:
       - name: Ensure generated shared i18n data
         run: node packages/app-core/scripts/ensure-shared-i18n-data.mjs
 
+      - name: Build core exports for real API harness
+        run: bun run --cwd packages/core build
+
       - name: Scenario runner unit coverage
         run: bun run --cwd packages/scenario-runner test -- src/executor.test.ts src/runtime-factory.test.ts src/scenario-pr-workflow.test.ts src/__tests__/deterministic-action-coverage.test.ts
 
@@ -708,6 +711,9 @@ jobs:
 
       - name: Ensure generated shared i18n data
         run: node packages/app-core/scripts/ensure-shared-i18n-data.mjs
+
+      - name: Build core exports for real API harness
+        run: bun run --cwd packages/core build
 
       - name: Install Playwright browsers (Chromium)
         run: .github/scripts/install-playwright-browsers.sh chromium

--- a/packages/app/e2e/accounts-ui/run-accounts-ui-e2e.mjs
+++ b/packages/app/e2e/accounts-ui/run-accounts-ui-e2e.mjs
@@ -135,12 +135,13 @@ const stubElizaCore = {
     }));
   },
 };
-// Swap ONLY the app-state barrel for the translator stub. The api barrel — the
-// network layer under test — stays real.
-const stubStateBarrel = {
-  name: "stub-state-barrel",
+// Swap only the app-state selectors for the translator stub. Account surfaces
+// import selectors from both the state barrel and its app-store module; the API
+// barrel — the network layer under test — stays real.
+const stubAppState = {
+  name: "stub-app-state",
   setup(b) {
-    b.onResolve({ filter: /^(\.\.\/)+state$/ }, () => ({
+    b.onResolve({ filter: /^(\.\.\/)+state(?:\/app-store)?$/ }, () => ({
       path: join(here, "accounts-fixture-state-stub.ts"),
     }));
   },
@@ -155,7 +156,7 @@ async function bundleFixture() {
     jsx: "automatic",
     loader: { ".tsx": "tsx", ".ts": "ts" },
     define: { "process.env.NODE_ENV": '"production"' },
-    plugins: [stubStateBarrel, stubElizaCore, stubNodeBuiltins],
+    plugins: [stubAppState, stubElizaCore, stubNodeBuiltins],
     write: false,
     absWorkingDir: repoRoot,
     logLevel: "silent",

--- a/packages/scripts/__tests__/scenario-pr-core-build-workflow.test.ts
+++ b/packages/scripts/__tests__/scenario-pr-core-build-workflow.test.ts
@@ -1,0 +1,54 @@
+/**
+ * Ensures real Scenario E2E API harnesses build the core package exports they
+ * resolve before starting local chat or accounts servers.
+ */
+import { describe, expect, test } from "bun:test";
+import { readFileSync } from "node:fs";
+
+interface WorkflowStep {
+  name?: string;
+  run?: string;
+}
+
+interface WorkflowJob {
+  steps?: WorkflowStep[];
+}
+
+interface Workflow {
+  jobs?: Record<string, WorkflowJob>;
+}
+
+const workflow = Bun.YAML.parse(
+  readFileSync(
+    new URL("../../../.github/workflows/scenario-pr.yml", import.meta.url),
+    "utf8",
+  ),
+) as Workflow;
+
+function expectCoreBuildBefore(jobName: string, harnessStepName: string): void {
+  const steps = workflow.jobs?.[jobName]?.steps ?? [];
+  const buildIndexes = steps.flatMap((step, index) =>
+    step.name === "Build core exports for real API harness" ? [index] : [],
+  );
+  const harnessIndex = steps.findIndex((step) => step.name === harnessStepName);
+
+  expect(buildIndexes).toHaveLength(1);
+  expect(steps[buildIndexes[0]]?.run).toBe("bun run --cwd packages/core build");
+  expect(harnessIndex).toBeGreaterThan(buildIndexes[0]);
+}
+
+describe("Scenario E2E real API build ordering (#20363)", () => {
+  test("builds core exports before the real local chat pipeline", () => {
+    expectCoreBuildBefore(
+      "scenario-unit-coverage",
+      "Real local chat pipeline (no model key, no llama)",
+    );
+  });
+
+  test("builds core exports before the real accounts UI API server", () => {
+    expectCoreBuildBefore(
+      "app-accounts-ui",
+      "Accounts UI browser e2e — real AccountList + real accounts routes + real AccountPool",
+    );
+  });
+});

